### PR TITLE
Merge development into main

### PR DESCRIPTION
**Pull request by axtloss**

### CODE REVIEW

File: src/functions/base.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ use std::path::PathBuf;
 use log::warn;
 
 pub fn install_base_packages(kernel: String) {
-    let mut kernel_to_install: String = String::new();
+    let kernel_to_install: String;
```
Comment (mchalunderscore) [inline src/functions/base.rs]: Is this actually necessary at all?

Comment (axtloss) [reply src/functions/base.rs]: I used this to verify that the kernel specified is correct, it's probably possible to do it differently

Comment (mchalunderscore) [reply src/functions/base.rs]: Looking at the code presented here, the value is never actually initialized, the type is just specified. I suggest that you do something like this for conditional variable assignment:

```rust
let kernel_to_install: String = if kernel.is_empty() {
    ...
} else {
    ...
};
```

Comment (axtloss) [reply src/functions/base.rs]: https://github.com/crystal-linux/jade/blob/acd5f935e651f40690bc7bcc59716f779ee061e7/src/functions/base.rs#L11-L24
this is the whole code, not sure if the if/else statement would work here

Comment (mchalunderscore) [reply src/functions/base.rs]: I've got it working, going to commit and push rn

**Comment by mchalunderscore** [comment]
That last commit also included a `cargo fmt`, do that more often please :sob:

**Comment by axtloss** [comment]
~~that's a thing?~~ Sure, I'll do that

**Comment by mchalunderscore** [comment]
LGTM

**Review by mchalunderscore — APPROVED** [review]

**Pull request merged**